src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -1492,6 +1547,70 @@ QList<QgsPointXY> QgsAdvancedDigitizingDockWidget::snapSegmentToAllLayers( const
   return segment;
 }
 
+bool QgsAdvancedDigitizingDockWidget::processCanvasPressEvent( QgsMapMouseEvent *event )
+{
+  if ( mCurrentTool )
+  {
+    mCurrentTool->canvasPressEvent( event );
```

Comment:
I think this should check:

   if ( event->isAccepted() )
      return;

elsewhere too, so that the subclasses can determine whether events should be propogated or not...